### ISSUE TITLE
Extend the Cartfile

### ISSUE THREAD

**Issue by sirghi** [Carthage/Carthage#2627]
Carthage is simple. So is this issue trying to be (although the first word in the title gives it away already).

Looking through the issues around here people seem to want to specify dependencies:
-  per scheme
-  per platform

Possible solutions:
- yaml
- ogdl
- prefixes and/or suffixes

While the discussions are long on this subject, I suggest we keep this short and focused, to decide in general whether we want this or not. I'll try to update the above lists if new ideas come up.

**Comment by KyleLeneau** [comment]
We should make sure that we can support a version of the file format / scheme. Cartfile’s now are only 1 version with no ability to extend and not break new/old versions. I’d like to see a version header be accounted for as well from the start so we can extend this further. The version header I am thinking of is similar to config file based CI systems like Travis or CircleCI `version: 0`. Maintaining backwards compatibility is important as well with the current files. Would supporting old and a new format be feasible?

**Comment by mdiep** [comment]
I'm not sure that's necessary. The minimum to build should always be the information that's currently in a `Cartfile`. We may add optional configuration, but the point here is to maintain compatibility with our existing format.

**Comment by sirghi** [comment]
Agree with @mdiep. Versioning would just complicate.

Also, I read people want to specify which schemes are valid to be built in general by Carthage (like for omitting test schemes and the likes). Should I add it to the list?

**Comment by mdiep** [comment]
I don't think we need to worry about building an exhaustive list of extensions that people want. :)
